packages/shared/src/components/search/SearchBarSuggestion.tsx

```diff
@@ -7,11 +7,51 @@ import {
   ButtonSize,
 } from '../buttons/Button';
 import { AiIcon } from '../icons';
+import { AnalyticsEvent, Origin, TargetType } from '../../lib/analytics';
+import { SearchSession } from '../../graphql/search';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
 
-export const SearchBarSuggestion = <TagName extends AllowedTags>({
+export type SuggestionOrigin =
+  | Origin.HomePage
+  | Origin.HistoryTooltip
+  | Origin.HistoryPage;
+type SearchBarSuggestionProps = ButtonProps<AllowedTags> & {
+  suggestion?: Partial<SearchSession>;
+  origin?: SuggestionOrigin;
+};
+
+export const SearchBarSuggestion = ({
   className,
+  origin,
   ...props
-}: ButtonProps<TagName>): ReactElement => {
+}: SearchBarSuggestionProps): ReactElement => {
+  const { trackEvent } = useContext(AnalyticsContext);
+
+  useEffect(() => {
+    if (props.suggestion?.id) {
+      trackEvent({
+        event_name: AnalyticsEvent.Impression,
+        target_type: TargetType.SearchRecommendation,
+        target_id: props.suggestion.id,
```

Comment:
using `id` here, but I wanted to point to where suggestions are loaded in `useSearchSuggestions` as it looks to me like the `id` is being dropped here? https://github.com/dailydotdev/apps/blob/f63091bfe9423bf3153f4ba16b6f6d45d0fd946d/packages/shared/src/hooks/search/useSearchSuggestions.ts#L26-L32

Reply:
In that case we'll need to add it right, to make this work?

Reply:
yep, I added it in this PR